digital_shadows_connector.py

```diff
@@ -22,61 +22,149 @@ class DigitalShadowsConnector(BaseConnector):
     def __init__(self):
         super(DigitalShadowsConnector, self).__init__()
 
+    def test_connectivity(self):
+        self.save_progress("Testing connectivity")
+        self.save_progress("test")
+        test_connectivity_connector = DSTestConnectivityConnector(self)
+        return test_connectivity_connector.test_connectivity()
+
+    def get_incident_by_id(self, param):
+        self.save_progress("Entered the function")
```

Comment:
For all the save_progress function you can give the function name or something related to the function